test/plugin/test_filter_parser.rb

```diff
@@ -0,0 +1,730 @@
+    d.instance_shutdown
+
+    d = create_driver(CONFIG_PASS_SAME_RECORD)
+    assert_nothing_raised {
+      d.run do
+        d.feed(@tag, Fluent::EventTime.now.to_i, {'foo' => 'bar'})
+      end
+    }
+    filtered = d.filtered
+    assert_equal 1, filtered.length
+    assert_nil filtered[0][1]['data']
+    assert_equal 'bar', filtered[0][1]['foo']
+  end
+
+  # suppress_parse_error_log test
+  INVALID_MESSAGE = 'foo bar'
+  VALID_MESSAGE   = 'col1=foo col2=bar'
+
+  def test_parser_error_warning
+    d = create_driver(CONFIG_INVALID_TIME_VALUE)
+    flexmock(d.instance.router).should_receive(:emit_error_event).
+      with(String, Integer, Hash, ParserError).once
+    d.run do
+      d.feed(@tag, Fluent::EventTime.now.to_i, {'data' => '{"time":[], "f1":"v1"}'})
+    end
+  end
+
+  CONFIG_DEFAULT_SUPPRESS_PARSE_ERROR_LOG = %[
+    key_name message
+    <parse>
+      @type regexp
+      expression /^col1=(?<col1>.+) col2=(?<col2>.+)$/
+    </parse>
+  ]
+
+  class DefaultSuppressParseErrorLogTest < self
+    setup do
+      # default(disabled) 'suppress_parse_error_log' is not specify
+      @d = create_driver(CONFIG_DEFAULT_SUPPRESS_PARSE_ERROR_LOG)
+    end
+
+    def test_raise_exception
+      flexmock(@d.instance.router).should_receive(:emit_error_event).
+        with(String, Integer, Hash, ParserError.new("pattern not match with data '#{INVALID_MESSAGE}'")).once
+      @d.run do
+        @d.feed(@tag, Fluent::EventTime.now.to_i, {'message' => INVALID_MESSAGE})
+      end
+    end
+
+    def test_nothing_raised
+      flexmock(@d.instance.router).should_receive(:emit_error_event).never
+      @d.run do
+        @d.feed(@tag, Fluent::EventTime.now.to_i, {'message' => VALID_MESSAGE})
+      end
+    end
+  end
+
+  CONFIG_DISABELED_SUPPRESS_PARSE_ERROR_LOG = %[
+    key_name message
+    suppress_parse_error_log false
+    <parse>
+      @type regexp
+      expression /^col1=(?<col1>.+) col2=(?<col2>.+)$/
+    </parse>
+  ]
+
+  class DisabledSuppressParseErrorLogTest < self
+    setup do
+      # disabled 'suppress_parse_error_log'
+      @d = create_driver(CONFIG_DISABELED_SUPPRESS_PARSE_ERROR_LOG)
+    end
+
+    def test_raise_exception
+      flexmock(@d.instance.router).should_receive(:emit_error_event).
+        with(String, Integer, Hash, ParserError.new("pattern not match with data '#{INVALID_MESSAGE}'")).once
+      @d.run do
+        @d.feed(@tag, Fluent::EventTime.now.to_i, {'message' => INVALID_MESSAGE})
+      end
+    end
+
+    def test_nothing_raised
+      flexmock(@d.instance.router).should_receive(:emit_error_event).never
+      @d.run do
+        @d.feed(@tag, Fluent::EventTime.now.to_i, {'message' => VALID_MESSAGE})
+      end
+    end
+  end
+
+  CONFIG_ENABELED_SUPPRESS_PARSE_ERROR_LOG = %[
+    key_name message
+    suppress_parse_error_log true
+    <parse>
+      @type regexp
+      expression /^col1=(?<col1>.+) col2=(?<col2>.+)$/
+    </parse>
+  ]
+
+  class EnabledSuppressParseErrorLogTest < self
```

Comment:
Migrated plugin code doesn't have `suppress_parse_error_log` option.
This test code is outdated, right?